lib/stack_master/config.rb

```diff
@@ -62,6 +62,21 @@ def unalias_region(region)
     end
 
     private
+    def load_template_compilers config
+      @template_compilers = {}
+      populate_template_compilers(config.fetch('template_compilers', {}))
+      merge_defaults_to_user_defined_compilers
+    end
+
+    def merge_defaults_to_user_defined_compilers
+      @template_compilers = default_template_compilers.merge(@template_compilers)
+    end
+
+    def populate_template_compilers user_defined_compilers
+      user_defined_compilers.each do |key, val|
```

Comment:
I've decided to have everything in the map as symbols. So it's more structured and also matches how we do things already.